### PR TITLE
Image: Fix caption link wrapping in wide/full alignment

### LINKED ISSUE #67206: Image Block: Caption with Link in Wide-Width and Full-Width Images Appears on Two Lines Instead of One Line
### Description

When a caption in an Image Block includes a link, and the image is set to either wide-width for full-width, the text and link in the caption are split up and shown on two lines.

When the Image Block is set to centered, the text and link appear on the same line.

Is this expected behavior for the Image Block? If so, is there an option in the toolbar or settings to remove the
when a link is used?

From what I see, the following CSS is causing the link being in one line: 
```
.wp-block-image.alignfull a, .wp-block-image.alignwide a {
    width: 100%;
}
```

### Step-by-step reproduction instructions

1. On a site with a block theme, such as Twenty Twenty-five, go to the page or post editor. 
2. Insert the Image Block and add an image.
3. Type in a caption such as "Photo Credit: Nature is Amazing Photo Collection".
4. Highlight the text "Nature is Amazing Photo Collection" and add a link. 
5. Change the Image Block alignment to Wide-Width or Full-Width
6. Notice the the text is split into two lines: 
"Photo Credit: 
Nature is Amazing Photo Collection" 
7. When inspecting the page, the HTML shows a <br> tag is added. 
8. Change the Image Block alignment to None
9. Notice that the text is not split and appears in one line: 
"Photo Credit: Nature is Amazing Photo Collection"

### Screenshots, screen recording, code snippet

Image

### Environment info

- WordPres 6.7.1
- GB: 19.7.0

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

### PR DESCRIPTION
**Pull request by himanshupathak95**
Fixes: #67206 

## What?
Changes the width CSS property of links within wide and full-width image block captions to prevent caption text from splitting across lines.

## Why?
Currently, when a caption in a wide or full-width Image Block contains a link, the linked text appears on a separate line from the rest of the caption text. This creates an undesirable visual break in what should be continuous caption text.

This occurs due to the `width: 100%` CSS property being applied to links within wide/full-width image blocks, forcing them to take up the full width and create a line break.

## Testing Instructions
1. Open the block editor with a block theme (e.g., Twenty Twenty-Five)
2. Insert an Image Block and add an image
3. Add a caption like "Photo Credit: Nature is Amazing Photo Collection"
4. Select "Nature is Amazing Photo Collection" and add a link
5. Test with Wide Width alignment: Caption should now appear on one line

### CODE REVIEW

File: packages/block-library/src/image/style.scss [hunk 1]
```diff
@@ -44,7 +44,7 @@
 
 	&.alignfull a,
 	&.alignwide a {
-		width: 100%;
+		width: auto;
```
Comment (carolinan) [inline packages/block-library/src/image/style.scss]: I believe this change needs to be limited to links inside captions, and not the linked image.

Comment (himanshupathak95) [reply packages/block-library/src/image/style.scss]: Hi @carolinan, Thanks for pointing this out.

I have addressed the feedback and now specifically targeting the caption width.

**Review by carolinan — APPROVED** [review]
I tested the change using Chrome and Firefox on Windows 11 and it is working well in both the editors and front. I tested with different themes, and also with different link settings such as the lightbox.

**Comment by t-hamano** [comment]
@himanshupathak95 Thanks for the PR.

I looked into this issue and it appears to have already been resolved by #67392.

I would like to close this PR, but thank you for your efforts.

**Pull request closed without merging**